### ISSUE TITLE
Attempted import error: 'signin' is not exported from 'next-auth/client' (imported as 'signin').

### ISSUE THREAD

**Issue by NishargShah** [nextauthjs/next-auth#1884]
## Description 🐜

I am getting an error when I login into the system and it works before, please check it

> Attempted import error: 'signin' is not exported from 'next-auth/client' (imported as 'signin').

Image: https://i.imgur.com/EMPECR9.png

## Code ☕️

```
import { signin } from 'next-auth/client';
......
const res = await signin('credentials', {
        redirect: false,
        email,
        password,
        remember,
});
```

## Environment 🖥

Please run this command:

```
$ npx envinfo --system --binaries --browsers --npmPackages "{next-auth}"
```

```
System:
    OS: Linux 5.8 Ubuntu 20.04.2 LTS (Focal Fossa)
    CPU: (8) x64 Intel(R) Core(TM) i5-8250U CPU @ 1.60GHz
    Memory: 160.81 MB / 15.42 GB
    Container: Yes
    Shell: 5.8 - /usr/bin/zsh
  Browsers:
    Chrome: 90.0.4430.85
    Firefox: 88.0

```

I will ready to help and contribute if it's a valid error.

**Comment by balazsorban44** [comment]
I am pretty sure `signIn` will work.

Looks like more and more people are having trouble with our default imports :confused::

https://github.com/nextauthjs/next-auth/blob/0eb4159737e66a15e2d5f2d9c3236117226d9baf/src/client/index.js#L358-L378

**Comment by balazsorban44** [comment]
I think it is because it makes things harder to tree-shake for bundlers.

Probably `import * as client from "next-auth/client"` and then `client.signin()` should work.?

Could you verify?

Also if `import {signIn} from "next-auth/client` works?

**Comment by balazsorban44** [comment]
Probably related to that Next.js started turning on Webpack 5 for people with simple next.config.js configurations.

**Comment by NishargShah** [comment]
I just found the solution, problem is the `case-sensitivity`

From

```
import { signin } from 'next-auth/client';
```

To

```
import { signIn } from 'next-auth/client';
```

so why it works before, I am using it from 2 months and today its show up why??

**Comment by balazsorban44** [comment]
See above.

I'll try to see if I can push a fix for this. But ultimately, we should just get rid of all the `export default {}` stuff, and have a single named version of functions. I don't see the reason to keep around `signin`.

**Comment by balazsorban44** [comment]
Could you maybe provide a reproduction so I can test against it?

**Comment by NishargShah** [comment]
> Could you please provide a reproduction so I can test against it?

I will provide you in some time

**Comment by NishargShah** [comment]
Here is the old version code sandbox snippet that I am using before

https://codesandbox.io/s/signin-with-next-auth3132-oytf9

Here is the new version code sandbox snippet

https://codesandbox.io/s/signin-with-next-auth3192-7qq1v

**Comment by balazsorban44** [comment]
@NishargShah I created a pre-release at `next-auth@3.19.5-beta.0` and tested it myself, and it seemed to work. Could you maybe please verify if your problems get resolved by it? :pray: If yes, then #1909 is safe to merge so it will be available in `latest`

**Comment by NishargShah** [comment]
I don't understand it, which is final `signIn` or `signin`?

because documentation said `signIn` and code say `signin`?

I am very confused.

**Comment by balazsorban44** [comment]
**BOTH** are currently supposed to be supported, for legacy reasons:

https://github.com/nextauthjs/next-auth/blob/dfe4620056b6ba3f72d1dd5ab166efb7f29db5c7/src/client/index.js#L369-L371

Thus, I am happy to fix this for now, so you could use either `signin` or `signIn`, but in a later breaking/major release, I think we should stop have two versions, as it is just confusing (as you present it here).

I did create a fix above, which would make it possible again for you to use `signin` as before, and I created a pre-release of it to test this out.

What I would like to ask from you is to check that if you are using `signin` in your code (as you did before), could please check if using the version `next-auth@3.19.5-beta.0` makes this bug go away?

**Comment by NishargShah** [comment]
@balazsorban44 I checked both are working fine :)

**Comment by balazsorban44** [comment]
thank you! 🙂

**Comment by balazsorban44** [comment]
Should be fixed in `3.19.5`! :tada:
